sdk/openai/Azure.AI.OpenAI/tests/OpenAIInferenceTests.cs

```diff
@@ -94,5 +110,32 @@ public void RequestFailedExceptionTest()
             var exception = Assert.ThrowsAsync<RequestFailedException>(async () => { await client.GetCompletionsAsync("BAD_DEPLOYMENT_ID", completionsRequest); });
             Assert.AreEqual(404, exception.Status);
         }
+
+        [RecordedTest]
+        public async Task StreamingCompletionsTest()
+        {
+            OpenAIClient client = GetClient();
+            CompletionsOptions requestOptions = new CompletionsOptions()
+            {
+                Prompt =
+                {
+                    "Tell me some jokes about mangos",
+                    "What are some disturbing facts about papayas?"
+                },
+                MaxTokens = 512,
+            };
+
+            using var fileWriter = File.CreateText("foo.txt");
+
+            fileWriter.WriteLine($"{Environment.TickCount} : START");
+            Response<StreamingCompletions> response = await client.GetStreamingCompletionsAsync(
+                CompletionsDeploymentId,
+                requestOptions);
+
+            await foreach (string message in response.Value.GetAsyncMessages())
+            {
+                fileWriter.WriteLine($"{Environment.TickCount} : {message}");
+            }
```

Comment:
Do we want to model disposing the response properly (even if it doesn't matter for this test)?

Reply:
Yeah, we don't want incorrect code to make it to the internet these days. OpenAI will learn wrong patterns from these tests, suggest them to unsuspecting humans who will soon be developing using copy-paste-oriented-programming, which in turn will make OpenAI not work properly :-)

Reply:
Agreed on all counts, but now I'm curious: is there (or should there be) an `IDisposable` `Response<T>` equivalent wrapper? Without one, it seems like we're relegated to the extremely clunky, old-style try/finally resource management pattern, i.e.
```csharp
Response<SomeDisposableType> response = client.GetResponse();
try
{
    // ... use response
}
finally
{
    response.Value?.Dispose();
}
```
Using the implicit `<T>` operator gets around this (e.g. `using SomeDisposableType responseValue = client.GetResponse()` will work) but if you want the `Response<T>` itself (e.g. for the raw payload) then I don't see a way to use `using` and all that goodness.

Reply:
For the moment, I updated the test to use/demonstrate the declare -> try -> finally pattern mentioned. I'm now curious about whether or not OpenAI would end up being smart enough to recommend that properly!

Reply:
When I discussed it with @tg-msft, we thought that for the preview response.Value would be disposable, so the called would do:
```csharp
using Model result = client.GetResponse().Value;
```

Reply:
After the preview, we will think about adding a disposable `Response<T>` variant/subtype. We don't want to make `Response<T>` disposable as most responses don't have to be disposed but if we implemented `IDisposable` on `Response<T>`, users would be compelled to dispose it.

Reply:
Makes sense -- and I for sure agree that having a (fabricated name) `public class DisposableResponse<T> : Response<T> where T : IDisposable` would be the way to go vs. changing any existing `Respose` type.

Applying `using` to the field of the top-level `Response<T>` still has some rough edges (inappropriate scoping of the `using` statement could cause a still-valid `Response<T>` to have a disposed/invalid `.Value` field) but I certainly like that better than declare/try/finally. I'll apply that in the test code.